python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -176,12 +180,11 @@ def from_assets(
 
     @property
     def all_asset_keys(self) -> AbstractSet[AssetKey]:
-        return self.materializable_asset_keys | self.source_asset_keys
+        return self._asset_dep_graph["upstream"].keys() | self.source_asset_keys
```

Comment:
Same result but I think this reads cleaner if it's `self.materializable_asset_keys | self.source_asset_keys`

Reply:
yeah good point re: readability

as for precomputing all_asset_keys as well, I think it's fine to wait and see if we hit any issues (this seems like a niche thing that only hits older python versions in a subset of cases, so would rather not take any additional memory hit if possible)